src/profile-logic/convert-markers.js

```diff
@@ -38,22 +38,18 @@ export function upgradeGCMinorMarker(marker: Object) {
 /*
  * Fix the units for GCMajor and GCSlice phase times.
  */
-function _upgradePhaseTimes(old_phases: Object): Object {
+export function convertPhaseTimes(old_phases: Object): Object {
   const phases = {};
   for (const phase in old_phases) {
     phases[phase] = old_phases[phase] * 1000;
   }
   return phases;
 }
 
-export function upgradeGCSliceMarker(marker: Object) {
-  if (marker.timings && marker.timings.times && !marker.timings.phase_times) {
-    marker.timings.phase_times = _upgradePhaseTimes(marker.timings.times);
-    delete marker.timings.times;
-  }
-}
-
-export function upgradeGCMajorMarker(marker: Object) {
+export function upgradeGCMajorMarker(
+  marker: Object,
+  upgrade_processed: boolean
```

Comment:
I'm not comfortable with this boolean, but maybe it's just a bad name ?

Reply:
TBH I don't completely understand why we have a different path.

Reply:
The boolean is true if we're upgrading the processed version rather than the gecko version.  I'd use an enum if JavaScript had a better type system.